lib/python/qmk/search.py

```diff
@@ -78,26 +86,17 @@ def search_keymap_targets(keymap='default', filters=[], print_vals=[]):
                     key = function_match.group('key')
                     value = function_match.group('value')
 
-                    if value is not None:
-                        if func_name == 'length':
-                            valid_keymaps = filter(lambda e, key=key, value=value: key in e[2] and len(e[2].get(key)) == int(value), valid_keymaps)
-                        elif func_name == 'contains':
-                            valid_keymaps = filter(lambda e, key=key, value=value: key in e[2] and value in e[2].get(key), valid_keymaps)
-                        else:
-                            cli.log.warning(f'Unrecognized filter expression: {function_match.group(0)}')
-                            continue
-
-                        cli.log.info(f'Filtering on condition: {{fg_green}}{func_name}{{fg_reset}}({{fg_cyan}}{key}{{fg_reset}}, {{fg_cyan}}{value}{{fg_reset}})...')
-                    else:
-                        if func_name == 'exists':
-                            valid_keymaps = filter(lambda e, key=key: key in e[2], valid_keymaps)
-                        elif func_name == 'absent':
-                            valid_keymaps = filter(lambda e, key=key: key not in e[2], valid_keymaps)
-                        else:
-                            cli.log.warning(f'Unrecognized filter expression: {function_match.group(0)}')
-                            continue
-
-                        cli.log.info(f'Filtering on condition: {{fg_green}}{func_name}{{fg_reset}}({{fg_cyan}}{key}{{fg_reset}})...')
+                    filter_fn = FUNCTIONS.get(func_name)
+                    if filter_fn is None:
+                        cli.log.warning(f'Unrecognized filter expression: {function_match.group(0)}')
+                        continue
```

Comment:
The old code also rejected attempts to use functions with a wrong number of arguments; the suggested code does not do that.  One way to fix that could be splitting `FUNCTIONS` into `UNARY_FUNCTIONS` and `BINARY_FUNCTIONS` and choosing one of them here based on `value is not None` (but lambdas in `UNARY_FUNCTIONS` would still need to have a dummy `v` argument, unless you move the call to `functools.partial()` here too).